Lib/slapdtest/_slapdtest.py

```diff
@@ -496,16 +535,36 @@ def _cli_auth_args(self):
                 '-D', self.root_dn,
                 '-w', self.root_pw,
             ]
-        return authc_args
+        return ['-H', ldap_uri] + authc_args
 
     # no cover to avoid spurious coverage changes
     def _cli_popen(self, ldapcommand, extra_args=None, ldap_uri=None,
                    stdin_data=None):  # pragma: no cover
         if ldap_uri is None:
             ldap_uri = self.default_ldap_uri
 
+        environ = os.environ.copy()
+
         if ldapcommand.split("/")[-1].startswith("ldap"):
-            args = [ldapcommand, '-H', ldap_uri] + self._cli_auth_args()
+            args = [ldapcommand] + self._cli_remote_args(ldap_uri)
+
+            # setting $LDAPNOINIT ignores all other environment and config files
+            environ.pop('LDAPNOINIT', None)
```

Comment:
This is a -1 from me because you would need to enumerate *all* libldap options here so that people's ldap.conf doesn't leak in as you rightly noticed. You'll need to use `-o` command-line options to provide tls settings (not in 2.4 unfortunately).

Reply:
I didn't notice `-o <opt>=<val>`.  I agree that this is preferable.  Any idea what the `<opt>` name is for `TLS_CACERT`?  I tried some variations on `-o tls-cacert=...` without success.

[`man ldapadd`](https://linux.die.net/man/1/ldapadd) gives an example of `-o nettimeout=...`, which I guess corresponds to `NETWORK_TIMEOUT` in [`ldap.conf`](https://linux.die.net/man/5/ldap.conf).  So there doesn't seem to be a programmatic mapping.

Reply:
In the OpenLDAP test suite, test068 uses a number of the -o options.  If you examine it you will find the following:

```
$LDAPSASLWHOAMI -o tls_cacert=$TESTDIR/tls/ca/certs/testsuiteCA.crt -o tls_reqcert=hard \
    -o tls_cert=$TESTDIR/tls/certs/bjensen@mailgw.example.com.crt -o tls_key=$TESTDIR/tls/private/bjensen@mailgw.example.com.key -ZZ -Y EXTERNAL -H $URIP1 \
    > $TESTOUT 2>&1
```

which indicates that an underscore is used as the separator rather than a hyphen.  Hope that helps!

Reply:
Also, the reason nettimeout is different is that it was implemented singularly before the global handling of options was added for 2.5. :/  Be nice to make it consistent, but that would break backwards compatibility.

Reply:
In case others find this.  As I read it, the authoritative list of option names in the openldap tree is [`libraries/libldap/init.c`](https://git.openldap.org/openldap/openldap/-/blob/07429efcff9dc50abb8ca265c8e8caa8cd25d01c/libraries/libldap/init.c#L80) with the ~legacy CLI-only synonyms in [`clients/tools/common.c`](https://git.openldap.org/openldap/openldap/-/blob/07429efcff9dc50abb8ca265c8e8caa8cd25d01c/clients/tools/common.c#L821).  Option names are matched with `strcasecmp()` (case insensitive) after [replacing](https://git.openldap.org/openldap/openldap/-/blob/07429efcff9dc50abb8ca265c8e8caa8cd25d01c/clients/tools/common.c#L815) '-' with '_'.  So it seems like `-o TLS_CACERT=...` and `-o tls-cacert=...` should be equivalent.

It isn't immediately clear to me why my initial guesses didn't seem to work.  I guess I had some other change which I didn't notice...  I'm making my first deep dive into the details of ldap, and am still wading through all of the gotchas.

Reply:
Right, `-o tls_cacert=` doesn't work for me because I'm still using the Debian [packaged](https://tracker.debian.org/pkg/openldap) openldap 2.4.57.  The expanded `-o` is added in 2.5.x.  So I think I'm going to scale back my ambitions.